api/src/controllers/user/postUser.test.ts

```diff
@@ -0,0 +1,253 @@
+});
+
+it('rejects invalid photoUrl', async () => {
+    req.body.photoUrl = 'htp//notaurl.com';
+
+    await postUser(req as Request, res as Response, next);
+
+    expect(next.mock.calls[0][0]).toMatchObject({ message: 'Invalid message body', status: 400, details: { photoUrl: 'Must be a valid url' } });
+});
+
+it('will not double register a user', async () => {
+    mockUserRepository.get.mockResolvedValueOnce([testConstants.user1]); // TODO
+
+    await postUser(req as Request, res as Response, next);
+
+    expect(mockAuth0Repository.giveUserRole).not.toBeCalled();
+    expect(userRepository.create).not.toBeCalled();
+    expect(res.status).toBeCalledWith(409);
+    expect(res.json).toBeCalledWith({ user: testConstants.user1 });
+});
+
+it('will not write to db if it fails to reach auth0', async () => {
+    const e = new InternalServerErrorException({}, new Error('test'));
+    mockUserRepository.get.mockResolvedValueOnce([]);
+    mockAuth0Repository.giveUserRole.mockRejectedValueOnce(e);
+
+    await postUser(req as Request, res as Response, next);
+
+    expect(userRepository.create).not.toBeCalled();
+    expect(next).toBeCalledWith(e);
+});
+
+it('works on the happy path', async () => {
+    mockUserRepository.get.mockResolvedValueOnce([]);
+    mockAuth0Repository.giveUserRole.mockResolvedValueOnce();
+    mockUserRepository.create.mockResolvedValueOnce([testConstants.user1, testConstants.userRole1]);
+
+    await postUser(req as Request, res as Response, next);
+
+    expect(res.status).toBeCalledWith(201);
+    expect(res.json).toBeCalledWith({ user: testConstants.user1 });
+});
+
+// it('works on happy path', async () => {
+//     const r: s.resume_reviews.JSONSelectable = {
+//         id: '52c2cbdc-e0a8-48e7-9302-92a37e016ab0',
+//         state: 'seeking_reviewer',
+//         created_at: '2021-06-07T04:51:55.717971+00:00',
+//         updated_at: '2021-06-07T04:51:55.717971+00:00',
+//         reviewee_id: '319ffe68-cac5-470f-9186-33371300c38f',
+//         reviewer_id: '97cf8fdc-884d-442a-ac71-9922b8f1ee5e',
+//     };
+//     const u: s.users.JSONSelectable = {
+//         id: '67e8ff47-8c31-4725-885c-e0e40455e7f5',
+//         email: 'email@domain.com',
+//         program: 'myccid',
+//         program: 'compe',
+//         year: 4,
+//         given_name: 'bob',
+//         family_name: 'saggit',
+//         full_name: 'bob saggit',
+//         photo_url: 'asdf.com',
+//         created_at: '2021-06-14T06:09:19.373404+00:00',
+//         updated_at: '2021-06-14T06:09:19.373404+00:00',
+//     };
+//     req.body = { reviewee: '67e8ff47-8c31-4725-885c-e0e40455e7f5' };
+//     mockUserRepository.get.mockResolvedValueOnce([u]);
+
+//     await postUser(req as Request, res as Response, next);
+
+//     expect(next).not.toBeCalled();
+//     expect(res.status).toBeCalledWith(201);
+//     expect(res.json).toBeCalledWith({ resumeReview: r });
+// });
```

Comment:
I believe this is covered in 217-220?

Reply:
LOLZ. Totally need to delete that. That was copy-pasta from another file when I was starting implementation.